packages/eslint-plugin-ckeditor5-rules/lib/rules/ckeditor-plugin-flags.js

```diff
@@ -0,0 +1,680 @@
+
+			// Eslint indent handling is a bit tricky. Only the first inserted method line is indented properly.
+			// The next ones need to be indented manually. The last one have have to be indented in both directions
+			// because eslint seems to be removing indention of the `pluginNameMethod` line.
+			const method =
+				`${ index ? '' : '\n\n' }${ indent }/**\n` +
+				`${ indent } * @inheritDoc\n` +
+				`${ indent } */\n` +
+				`${ indent }public static override get ${ requiredFlag.name }(): ${ requiredFlag.returnValue } {\n` +
+				`${ indent }\treturn ${ requiredFlag.returnValue };\n` +
+				`${ indent }}${ isLastIndex ? '' : '\n\n' }`;
+
+			return fixer.insertTextAfter( pluginNameMethod, method );
+		} );
+
+		return [
+			...removedMethodsFixes,
+			...injectedMethodsFixes
+		];
+	};
+}
+
+/**
+ * Creates fixer for removing class methods by their names.
+ *
+ * @param {Object} context
+ * @param {Object} classDeclaration
+ * @param {Array.<String>} names
+ * @returns {Function}
+ */
+function createRemoveClassMembersFixer( context, classDeclaration, names ) {
+	// Cache last removed new line to avoid overlapping ranges during removal of multiple methods
+	let lastRemovedNewLine = 0;
+
+	// Remove all methods that are named like the flags.
+	return fixer => {
+		const ranges = getAllClassMembers( classDeclaration )
+			.filter( element => names.includes( element.key?.name ) )
+			.map( method => {
+				const sourceCode = context.getSourceCode();
+				const commentsBeforeMethod = sourceCode.getCommentsBefore( method );
+
+				const endRange = sourceCode.text.indexOf( '\n', method.range[ 1 ] ) + 1;
+				let startRange = sourceCode.text.lastIndexOf( '\n', commentsBeforeMethod[ 0 ]?.range[ 0 ] || method.range[ 0 ] );
+
+				// Prevent incorrect formatting of the methods that have no space between them.
+				// It prevents eating new line from the previous method.
+				if ( sourceCode.text[ startRange - 1 ].trim() === '}' ) {
+					startRange++;
+				}
+
+				// It prevents overlapping ranges when removing multiple methods.
+				// From time to time endline from previous and current method are the same which causes eslint to throw an error.
+				if ( startRange < lastRemovedNewLine ) {
+					startRange = lastRemovedNewLine + 1;
+				}
+
+				lastRemovedNewLine = endRange;
+
+				return [ startRange, endRange ];
+			} );
+
+		// Detect overlapping ranges before creating fixes. It's better to report a warning
+		// than crashing whole eslint process. This usually happens when the parser struggles with
+		// removing multiple methods at once that share the same endline. It should be handled in the code above
+		// but it's better to have a safety net here.
+		for ( let i = 0; i < ranges.length - 1; i++ ) {
+			// It looks like ranges overlap lets abort the fixer.
+			if ( ranges[ i ][ 1 ] > ranges[ i + 1 ][ 0 ] ) {
+				return [];
+			}
+		}
+
+		return ranges.map( range => fixer.removeRange( range ) );
+	};
+}
```

Comment:
This complicates this already complicated lint rule so I'd suggest to remove it.